### ISSUE TITLE
[Feature Request 1.18.1] Interact/Use Quest Type 

### ISSUE THREAD

**Issue by itsdinkd** [lorddusk/HQM#634]
Interact Quest type -- Currently, there is no way to give the quest maker the ability to make a quest that depends on interaction ("use") a item. An example would be player uses waystones. Without this, it's hard to give a quest for a player to use a waystone as in my modpack, they are something that isn't easily craftable, but are something that are spread out through the world that they can use. This could be used for many other things that the player is required to interact with that they cannot have in their inventory, mainly in a modded environment. Like using certain machines, or interacting with crops, fluid tanks, teleporters, rings, the list goes on. (I'm thinking how project mmo does this might be beneficial to look at if needed)

**Comment by kirderf1** [comment]
Each request should normally have its own issue, so that the requests can be hadled separately, and that the relevant issue can be closed once a feature is implemented.

**Comment by kirderf1** [comment]
The second feature request sounds like a duplicate of #421. If you agree, I'd suggest rewording this issue to focus solely on your first feature request, and add a comment to #421 if there's something not already covered over there in regards to the second request.
